src/agent/src/rpc.rs

```diff
@@ -258,8 +258,6 @@ impl AgentService {
             // Find the sandbox storage used by this container
             let mounts = sandbox.container_mounts.get(&cid);
             if let Some(mounts) = mounts {
-                remove_mounts(mounts)?;
-
                 for m in mounts.iter() {
```

Comment:
@Burning1020 - I'm not sure this is correct - see https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/rpc.rs#L159..L165

/cc @sameo, @yyyeerbo since they have touched this part of the code.

tbh, it would be great if we could add some [unit tests](https://github.com/kata-containers/tests/blob/main/Unit-Test-Advice.md#rust-table-driven-tests) for this functionality to make the behaviour totally clear.

Reply:
plz see that the contaier mounts were removed in the func `unset_and_remove_sandbox_storage` again, and all mounts in `sandbox.container_mounts` must be contained in `sandbox.storages` as it all comes from `add_storages`.
/cc @sameo, @yyyeerbo

Reply:
Having a unit test here would definitely help clarify, though I admit, it'll be a larger test with a lot of setup.

ie, we'd want to:
```
create_sandbox
do_create_container
do_remove_container
 ** check mounts **
destory_sandbox
 ** check all mounts are gone **
 ```
 
 In this particular case, though, I agree -- we are checking and of applicable removing the mount(s0 as part of the unset_and_remove_storage() a few lines down. A unit test will help clarify. @Burning1020 I appreciate your help; is this something you'd be able to help push?

Reply:
@egernst OK, i will try some ut. Well, you know, hard to work with RUST for me. Anyway, plz review it more carefully after my new push.

Reply:
Thanks @Burning1020 - please could you ensure that test only runs as `root` as we run the tests as `root` *and* non-`root`.